src/main.py

```diff
@@ -72,19 +71,22 @@ def run():
     else:
         llm: ILLM = OllamaLLM()
         code_condition: ICondition = CodeCondition(llm)
+        size_condition: ICondition = SizeCondition(num_chars=500)
         open_source_api = api_provider.get(config.get("OPEN_SOURCE_API"), **config)
         closed_source_api = api_provider.get(config.get("CLOSED_SOURCE_API"), **config)
-        router = AdvancedRouter(code_condition, open_source_api, closed_source_api)
+        router = AdvancedRouter([code_condition, size_condition], open_source_api, closed_source_api)
 
-    # Redirect tqdm output to logger
+    batch_size = int(config["BATCH_SIZE"])
+    total_batches = (len(dataset_loader.dataset) + batch_size - 1) // batch_size
     tqdm_out = TqdmToLogger(logger)
-    for i, row in tqdm(dataset.df.iterrows(), total=dataset.df.shape[0], file=tqdm_out):
 
+    for i, batch in enumerate(tqdm(dataset_loader.dataset.iter(batch_size=batch_size), total=total_batches, file=tqdm_out)):
+        batch_df = pd.DataFrame(batch)
+        batch_df.index = range(i * batch_size, i * batch_size + len(batch_df))
```

Comment:
Create a local variable here to move `i * batch_size` to a variable called offset.